### DISCUSSION TITLE
Dynamic siteTitle

### DISCUSSION THREAD

**Discussion by mammadataei** [vuejs/vitepress#1573] [Q&A]
Hi, is it possible to change the `siteTitle` for each sub-directory?! Something like [multiple sidebars](https://vitepress.vuejs.org/guide/theme-sidebar#multiple-sidebars) but for the site title.

**Comment by lieonlion** [comment]
I also want to be able to change the title in the top left to something different depending on the current page path, any way to do this?
![image](https://github.com/user-attachments/assets/48189cb0-a53e-4abb-bb3d-2733cd55da86)

**Comment by lemonicy** [comment]
You can use vue watchers.

Use this in your common template:

```
import { watch, nextTick } from 'vue';
import { useData, useRouter } from 'vitepress';
import { changeHeader } from { PATH_OF_CHANGEHEADER_FUNCTION };

const { frontmatter } = useData();
const router = useRouter();

watch(() => router.route.data.relativePath, (path) => {
    nextTick(() => {
        changeHeader(frontmatter.value.title);
    }, 100);
}, { immediate: false });
```

For Maintainability, you can put this in another file:

```
export function changeHeader(customTitle) {
    document.head.querySelector("title").innerText = customTitle;
}
```

In addition, if you need to change meta description or something else you want, you can refer to this.
